src/upload/getFileSize.handler.ts

```diff
@@ -0,0 +1,31 @@
+import { loggingFactory } from '../logger'
+import { ProviderManager } from '../providers'
+
+const logger = loggingFactory('getFileSize.handler')
+type GetFileSizeRouteHandler = (req: any, res: any) => Promise<void>
+
+export default function (storageProvider: ProviderManager): GetFileSizeRouteHandler {
+  return async (req: any, res: any): Promise<void> => {
+    const { hash } = req.query
+    const missedParams = ['hash'].filter(k => !req.query[k])
+
+    if (missedParams.length) {
+      return res.status(422).json({
+        error: `Params ${missedParams} required`
+      })
+    }
+
+    logger.info(`Retrieving file size for hash ${hash}`)
+    try {
+      return res.json({
+        fileHash: hash,
+        fileSizeBytes: await storageProvider.getActualFileSize(hash)
```

Comment:
Hmmm I don't think this is a good idea. The `ipfs dag stat` downloads all the data locally, so if we are talking about some big data, this will certainly timeout. I would use the `object stat` call here.

Reply:
Ok

Reply:
Then looks like we don't need `dag.stat` at all

Reply:
Yeah I don't think so here